storage/engine/rocksdb.go

```diff
@@ -428,7 +364,20 @@ func (r *RocksDB) NewIterator() Iterator {
 	return newRocksDBIterator(r.rdb, nil)
 }
 
-// Returns a new Batch wrapping this rocksdb engine.
+// NewSnapshot creates a snapshot handle from engine and returns a
+// read-only rocksDBSnapshot engine.
+func (r *RocksDB) NewSnapshot() Engine {
+	if r.rdb == nil {
+		log.Errorf("RocksDB is not initialized yet")
```

Comment:
In general I think it's better to panic than to log an error and return nil (when we cannot return an `error`) (here and inMemSnapshot.New{Batch,Snapshot})